### PR TITLE
[RLlib] Issue 42385: Fix `torch|tf_utils` explained_variance functions.

### LINKED ISSUE #42385: [RLlib] Missing "()" in function "explained_variance()" causing division by zero error.
### What happened + What you expected to happen

The return of the "explained_variance()" in the "ray/rllib/utils/torch_utils.py" is missing "()".     Change[0]" to "return torch.max(min_, 1 - (diff_var / (y_var + SMALL_NUMBER))[0]"

### Versions / Dependencies

ray 2.9.0
python3.10
ubuntu

### Reproduction script

def explained_variance(y: TensorType, pred: TensorType) -> TensorType:
    """Computes the explained variance for a pair of labels and predictions.

    The formula used is:
    max(-1.0, 1.0 - (std(y - pred)^2 / std(y)^2))

    Args:
        y: The labels.
        pred: The predictions.

    Returns:
        The explained variance given a pair of labels and predictions.
    """
    y_var = torch.var(y, dim=[0])
    diff_var = torch.var(y - pred, dim=[0])
    min_ = torch.tensor([-1.0]).to(pred.device)
    return torch.max(min_, 1 - (diff_var / (y_var + SMALL_NUMBER)))[0]

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by sven1977**
Issue 42385: Fix `torch|tf_utils` explained_variance functions.

## Why are these changes needed?

Closes #42385 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/torch_utils.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def explained_variance(y: TensorType, pred: TensorType) -> TensorType:
     y_var = torch.var(y, dim=[0])
     diff_var = torch.var(y - pred, dim=[0])
     min_ = torch.tensor([-1.0]).to(pred.device)
-    return torch.max(min_, 1 - (diff_var / y_var + SMALL_NUMBER))[0]
+    return torch.max(min_, 1 - (diff_var / (y_var + SMALL_NUMBER)))[0]
```
Comment (simonsays1980) [inline rllib/utils/torch_utils.py:313]: Or using `torch.finfo().eps`?

File: rllib/utils/tf_utils.py [hunk 2]
```diff
@@ -94,7 +95,7 @@ def explained_variance(y: TensorType, pred: TensorType) -> TensorType:
     """
     _, y_var = tf.nn.moments(y, axes=[0])
     _, diff_var = tf.nn.moments(y - pred, axes=[0])
-    return tf.maximum(-1.0, 1 - (diff_var / y_var))
+    return tf.maximum(-1.0, 1 - (diff_var / (y_var + SMALL_NUMBER)))
```
Comment (simonsays1980) [inline rllib/utils/tf_utils.py:98]: Or using `tf.keras.backend.epsilon()`?

**Review by simonsays1980 — APPROVED** [review]
LGTM. Small fixes.

**Review by kouroshHakha — APPROVED** [review]
